modules/jenkins-plugin/src/main/java/com/smartbear/ready/jenkins/RemoteTestStarter.java

```diff
@@ -0,0 +1,539 @@
+            executionRequest.withCustomProperties(null, projectProperties);
+        }
+        return (TestServerExecution) executor.executeProject(executionRequest.build());
+    }
+
+    private TestServerExecution executeRepositoryProject(TestServerClient client) {
+        ProjectExecutor executor = client.createProjectExecutor();
+        Map<String, String> projectProperties = getProjectProperties();
+        Set<String> tagNames = buildStringSet(tags);
+        RepositoryProjectExecutionRequest.Builder executionRequest = RepositoryProjectExecutionRequest.Builder.forProject(pathToProjectFile)
+                .fromRepository(repositoryName)
+                .forTestSuite(testSuiteName)
+                .forTestCase(testCaseName)
+                .forEnvironment(environment)
+                .forTags(tagNames)
+                .withProjectPassword(projectFilePassword)
+                .withEndpoint(hostAndPort);
+        if (!projectProperties.isEmpty()) {
+            executionRequest.withCustomProperties(null, projectProperties);
+        }
+        return (TestServerExecution) executor.executeRepositoryProject(executionRequest.build());
+    }
+
+    public String getPathToProjectFile() {
+        return pathToProjectFile;
+    }
+
+    public String getTestCaseName() {
+        return testCaseName;
+    }
+
+    public String getTestSuiteName() {
+        return testSuiteName;
+    }
+
+    public String getServerUrl() {
+        return isNotBlank(serverUrl) ? serverUrl : getDescriptor().getServerUrl();
+    }
+
+    public String getUserName() {
+        return isNotBlank(userName) ? userName : getDescriptor().getUserName();
+    }
+
+    public String getPassword() {
+        return isNotBlank(password) ? password : getDescriptor().getPassword();
+    }
+
+    public String getEnvironment() {
+        return environment;
+    }
+
+    public String getHostAndPort() {
+        return hostAndPort;
+    }
+
+    public String getTags() {
+        return tags;
+    }
+
+    public String getProjectFilePassword() {
+        return projectFilePassword;
+    }
+
+    public String getTestType() {
+        return testType.toString();
+    }
+
+    public String getExtraFiles() {
+        return extraFiles;
+    }
+
+    public String getReportFile() {
+        return reportFile;
+    }
+
+    private Map<String, String> getProjectProperties() {
+        if (isBlank(projectProperties)) {
+            return Collections.emptyMap();
+        } else {
+            Map<String, String> returnValue = new HashMap<>();
+            Properties properties = new Properties();
+            try {
+                properties.load(new StringReader(projectProperties));
+                for (Object key : properties.keySet()) {
+                    String keyString = String.valueOf(key);
+                    returnValue.put(keyString, properties.getProperty(keyString));
+                }
+            } catch (IOException ignore) {
+
```

Comment:
It's a StringReader so IOException will never be thrown. But I'll add a comment.